### PR TITLE
"acceptance" tests

### LINKED ISSUE #185: Python API 1.0  acceptance tests are in place and can be executed easily.
It is left to the discretion of the engineer working on this to assess which ones should also be converted to unit tests. Please reach out to me if you'd like to discuss 

Initial candidate of tests:

- [ ] For both organisms ,full `obs`, `var` and `X` can be iterated e.g. `for chunk in ["obs"].read(): ...` 
- [ ]  For both organisms, full `obs` and `var` can be loaded in a pandas.DataFrame e.g. `["obs"].read().concat().to_pandas`
- Experiment query operations can be done over a large slice of data, e.g. with `.ms["RNA"].axis_query(tiledbsoma.AxisQuery(value_filter = "tissue == 'brain')` then perform to completion the main `ExperimentQuery` operations:
   - [ ] `for chunk in obs(): ...`
   - [ ] `for chunk in var(): ...`
   - [ ] `for chunk in X().tables(): ...`
- Large AnnDatas can be loaded, document computing requirements:
   - [ ] largest possible in <20 min @bkmartinjr has made it happened in an EC2 instance `r6id.32xlarge` `get_anndata(census = census, organism = "Homo sapiens",)`
   - [ ] most common cell type `get_anndata(census = census, organism = "Homo sapiens", obs_value_filter = "cell_type == 'neuron'")`  
   - [ ] most common tissue `get_anndata(census = census, organism = "Homo sapiens", obs_value_filter = "cell_type == 'brain'")`

### PR DESCRIPTION
**Pull request by bkmartinjr**
Fixes #185 

Notes to reviewer:
* all tests added as pytest functions
* the "expensive" tests are marked as such, and several require very large host memory to complete
* added a README with details on running the tests and as a place to record historical runs for comparison

### CODE REVIEW

File: api/python/cell_census/pyproject.toml [hunk 1]
```diff
@@ -65,7 +65,8 @@ plugins = "numpy.typing.mypy_plugin"
 
 [tool.pytest.ini_options]
 markers = [
-    "live_corpus: runs on the live Cell Census data corpus",
+    "live_corpus: runs on the live Cell Census data corpus and small enough to run in CI",
+    "expensive: too expensive to run regularly or in CI",
```
Comment (atolopko-czi) [inline api/python/cell_census/pyproject.toml:69]: 👍

File: api/python/cell_census/tests/README.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+You can also combine them, e.g.,
+
+> pytest -m 'not live_corpus' --expensive
+
+# Acceptance (expensive) tests
+
+These tests are periodically run, and are not part of CI due to their overhead.
+
+Please record:
```
Comment (atolopko-czi) [inline api/python/cell_census/tests/README.md:38]:
```suggestion
When run, please record the results below and commit to git:
```

File: api/python/cell_census/tests/test_acceptance.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        assert table_iter_is_ok(
+            census["census_data"][organism].ms["RNA"].var.read(column_names=["soma_joinid", "feature_id"])
+        )
+        assert table_iter_is_ok(census["census_data"][organism].ms["RNA"].X["raw"].read().tables())
+
+
+@pytest.mark.live_corpus
+@pytest.mark.parametrize("organism", ["homo_sapiens", "mus_musculus"])
+@pytest.mark.parametrize(
+    "obs_value_filter", ["tissue=='aorta'", pytest.param("tissue=='brain'", marks=pytest.mark.expensive)]
```
Comment (atolopko-czi) [inline api/python/cell_census/tests/test_acceptance.py:84]: pytest is great; TIL: param values can add marks

File: api/python/cell_census/tests/README.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+- date
+- host / instance type
+- Python & package versions and OS (tip: use tiledbsoma.show_package_versions())
+- full output of: `pytest --durations=0 --expensive ./api/python/cell_census/tests/`
```
Comment (atolopko-czi) [inline api/python/cell_census/tests/README.md:43]: also, the cell census release, unless that shows up in the output?  motivation is to know what data size tests last passed for. Maybe each test could output relevant sizes on full reads, like obs.  Or output len(obs), len(var), X.nnz once.

**Review by atolopko-czi — APPROVED** [review]
Tests look great. Two minor suggestions to consider before merging.

**Review by ebezzi — APPROVED** [review]
Looks great!

File: api/python/cell_census/tests/README.md (hunk 4 above)
Comment (bkmartinjr) [reply api/python/cell_census/tests/README.md:44]: yes on recording the version aliased to `latest`

I'm not sure we want the tests generating a bunch of output unless it is regularly used.  I think your census version is the ideal solution.

File: api/python/cell_census/tests/test_acceptance.py (hunk 3 above)
Comment (bkmartinjr) [reply api/python/cell_census/tests/test_acceptance.py:84]: always pays to RT*M :-)

**Pull request merged**